code/stable/gooltest/cpp/HelloWorld/HelloWorld.cpp

```diff
@@ -126,33 +126,31 @@ int main(int argc, const char *argv[]) {
     mySlicedList7 = temp5;
     
     vector<double> temp6(0);
-    for (int i4 = 3; 0 < i4 && i4 <= 3 && z < 0; i4 += z) {
+    for (int i4 = 3; i4 < 0; i4 += z) {
```

Comment:
I would not be surprised if this change was due to your defaulting some things to 0!

It is also strange that this changed, as you don't have any code for simplifying constraints / boolean expressions. So I really suspect it's the defaults.

Reply:
These should be the relevant lines of code:
https://github.com/JacquesCarette/Drasil/blob/e72dd927fbcc94ffdf0eacda6252826d9b74bcff/code/drasil-gool/lib/Drasil/GOOL/LanguageRenderer/Macros.hs#L98-L109

I believe @JacquesCarette was right: setting values to `litInt 0` makes the line `100` case trigger instead of the line `101` case.

Reply:
In this case, where `step` is a variable with a negative value, `valueInt stepV` will return `Nothing`, making `mbStepV = Nothing`, which I believe would trigger the logic starting at line 101. However, in this code, how does the default value `litInt 0` affect `mbStepV` and result in `mbStepV` getting a value of `0`, which subsequently generates the code `for (int i4 = 3; i4 < 0; i4 += z)`?

Reply:
This is the line creating the `listSlice` in the example:
https://github.com/JacquesCarette/Drasil/blob/e72dd927fbcc94ffdf0eacda6252826d9b74bcff/code/drasil-code/test/HelloWorld.hs#L170-L173

I believe that the specific reason that line `100` is triggering instead of line `101` is that the value of `step` uses `valueOf`, which gives `litInt 0` as its literal value.

Variables aren't literal values; they hold values.  Since GOOL doesn't know what the value they're holding is, the correct `litVal` for them should be `Nothing`, as they aren't literal values.